core/node/bitswap.go

```diff
@@ -35,6 +35,7 @@ func OnlineExchange(cfg *config.Config, provide bool) interface{} {
 
 		opts := []bitswap.Option{
 			bitswap.ProvideEnabled(provide),
+			bitswap.ProviderSearchDelay(0), // See https://github.com/ipfs/go-ipfs/issues/8807 for rationale
```

Comment:
Shouldn't it be more like 20ms to avoid doing DHT searches on files we already have working sessions ?

Actually this check seems not like what we want. We would want to scale this delay per how good the current session is doing.

A dead session with no active peer should be 0. But a sessions with 3 peers and all blocks are downloaded really fast mean you are bottlenecked elsewhere and don't need to run DHT searches.

Reply:
This setting is currently a brute force approach without any adaptive behaviour. To build that adaptive algorithm we'd need to characterize what metrics should be kept and used to influence the delay. What precisely would we mean by "downloaded really fast", for example.